lib/datadog_backup/resources.rb

```diff
@@ -2,175 +2,131 @@
+      @body = sanitize(body)
     end
 
-    def initialize(options)
-      @options = options
-      @banlist = []
-      ::FileUtils.mkdir_p(mydir)
-    end
-
-    def myclass
-      self.class.to_s.split(':').last.downcase
-    end
-
-    # Create a new resource in Datadog
-    def create(body)
+    def create(api_resource_name: @api_resource_name)
       headers = {}
-      response = api_service.post("/api/#{api_version}/#{api_resource_name}", body, headers)
-      body = body_with_2xx(response)
-      LOGGER.warn "Successfully created #{body.fetch(id_keyname)} in datadog."
-      LOGGER.info 'Invalidating cache'
-      @get_all = nil
+      body = @api_service.post_body("/api/#{@api_version}/#{api_resource_name}", @body, headers)
+      @id = body[@id_keyname]
+      LOGGER.warn "Successfully created #{@id} in datadog."
+      self.class.invalidate_cache
       body
     end
 
-    # Update an existing resource in Datadog
-    def update(id, body)
+    def update(api_resource_name: @api_resource_name)
       headers = {}
-      response = api_service.put("/api/#{api_version}/#{api_resource_name}/#{id}", body, headers)
-      body = body_with_2xx(response)
-      LOGGER.warn "Successfully restored #{id} to datadog."
-      LOGGER.info 'Invalidating cache'
-      @get_all = nil
+      body = @api_service.put_body("/api/#{@api_version}/#{api_resource_name}/#{@id}", @body, headers)
+      LOGGER.warn "Successfully restored #{@id} to datadog."
+      self.class.invalidate_cache
       body
     end
 
-    # If the resource exists in Datadog, update it. Otherwise, create it.
-    def restore(id)
-      body = load_from_file_by_id(id)
+    def restore
+      @body = body_from_backup
       begin
-        update(id, body)
+        update
       rescue RuntimeError => e
         raise e.message unless e.message.include?('update failed with error 404')
 
-        create_newly(id, body)
+        create_newly
+      ensure
+        @body
       end
     end
 
-    # Return the Faraday body from a response with a 2xx status code, otherwise raise an error
-    def body_with_2xx(response)
-      unless response.status.to_s =~ /^2/
-        raise "#{caller_locations(1,
-                                  1)[0].label} failed with error #{response.status}"
-      end
-
-      response.body
-    end
-
     private
 
-    def api_url
-      ENV.fetch('DD_SITE_URL', 'https://api.datadoghq.com/')
+    # Create a new resource in Datadog, then move the old file to the new resource's ID
+    def create_newly
+      delete_backup
+      create
+      backup
     end
 
-    def api_version
-      raise 'subclass is expected to implement #api_version'
+    # Returns a hash with @banlist elements removed
+    def except(hash)
+      outhash = hash.dup
+      @banlist.each do |key|
+        outhash.delete(key) # delete returns the value at the deleted key, hence the tap wrapper
+      end
+      outhash
     end
```

Comment:
This method could be more concisely implemented as:

    @banlist.each_with_object(hash.dup) do |key, outhash|
      outhash.delete(key)
    end